Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Utilities/RectTransformCubeScaler.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Utilities
+{
+    /// <summary>
+    /// RectTransforms do not scale 3d objects (such as unit cubes) to fit within their bounds.
+    /// This helper class will apply a scale to fit a unit cube into the bounds specified by the RectTransform.
+    /// The Z component is scaled to the min of the X and Y components.
+    /// </summary>
```

Comment:
I'm still not sure where I would use this, could you provide an example of when I'd want to use it, and where it's used in MRTK? I don't see where it's used in this PR but perhaps I am missing something.

Reply:
I use it in a prefab.  Specifically, I needed it for [PressableButtonHololens2UnityUI.prefab](https://github.com/AdamMitchell-ms/MixedRealityToolkit-Unity/blob/unityui-nearfield-improvements2/Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Prefabs/PressableButtonHololens2UnityUI.prefab).

![image](https://user-images.githubusercontent.com/41760870/64275608-d3149c80-cefa-11e9-86c5-f67c3e107782.png)

PressableButtonHololens2 needs a cube for the compressable visual.  But in Unity, a cube ignores a RectTransform's Width/Height.  So to get it to fit into the bounds specified by the size, I created this little script.

To see this for yourself, create a canvas in Unity.  Then place an image in it and change the Width/Height of the rect transform.  The image will be resized to fit inside that size.  Now create a cube and give it a rect transform.  Changing the width/height has no effect on the cube.  The cube's size is determined only by the Scale of the transform hierarchy.  So this script takes the rect transform's width/height and applies a scale which will cause a 1x1x1 sized object to fit within the bounds.